### DISCUSSION TITLE
FocusManager can't get the usercontrol after an Focus()

### DISCUSSION THREAD

**Discussion by Tenjim** [AvaloniaUI/Avalonia#16903] [Q&A] [answered]
Hi guys,

I have one question about Focusmanager. I'm on Avalonia 11.1.3. After an Focus() on a UserControl, I don't have that UserControl when I get the  topLevel.FocusManager.GetFocusedElement().

There is an option to force the focus on an UserControl ?

**Comment by stevemonaco** [comment]
Make sure the `UserControl` has `Focusable="True"`. The default should be `False` which means it can't be focused.

**Reply by Tenjim** [reply]
I have check that yeah but the focusmanager (GetFocusedElement()) don't get me the right control after an Focus().

**Reply by stevemonaco** [reply]
It works perfectly fine for me, so I have no idea what you're doing.

MainWindow:
```xml
<Window ...>
    <Grid RowDefinitions="auto, *">
        <Button Click="Focus_Click" Content="Focus TestControl" />
        <v:TestControl Grid.Row="1" x:Name="testControl" />
    </Grid>
</Window>
```
```cs
private void Focus_Click(object? sender, RoutedEventArgs e)
{
    testControl.Focus();
    var c = FocusManager.GetFocusedElement();
}
```

UserControl (TestControl):
```xml
<UserControl ... Focusable="True">
    Welcome to Avalonia!
</UserControl>
```

![image](https://github.com/user-attachments/assets/0c5b3821-6952-45d7-9a69-92dcfd166963)

**Reply by Tenjim** [reply]
I have just investigate that case : 
when I check for IsEffectivelyVisible property on the control. I put the focus and check the focusmanager but the focus don't set this control. The only way to have my focus when my control is loaded is to doing focus when there is the event AttachedToVisualTree.

Maybe I missunderstood IsEffectivelyVisible ? For me it's when is loaded no ?

**Reply by stevemonaco** [reply, marked as answer]
`IsEffectivelyVisible` doesn't have anything specifically to do with focus, besides the fact that you can't focus an invisible control. It allows a parent control with `IsVisible="False"` to propagate that state to hide their children, as well.

**Reply by Tenjim** [reply]
Ah thank, yeah, it's more clear for me :)
